onnxruntime/core/providers/cuda/cu_inc/common.cuh

```diff
@@ -208,8 +208,9 @@ static INT CeilDiv(INT a, INT2 b)  // ceil(a/b)
 
 struct GridDim {
   enum : CUDA_LONG {
-    maxThreadsPerBlock = 1024,  // use this many threads per block
-    maxWarpsPerBlock = 32,      // use this many warps per block. This means 1024 threads for warpSize=32
+    maxThreadsPerBlock = 256,  // max threads per block
+    maxWarpsPerBlock = 32,     // max warps per block
+    maxElementsPerThread = 4,  // max element processed per thread
```

Comment:
As described below, `GridDim` is designed to be used specifically as kernel launch config. Therefore, `maxElementsPerThread` is not supposed to be here.

Reply:
It's part of the kernel launch.

---
In reply to: [343290755](https://github.com/microsoft/onnxruntime/pull/2335#discussion_r343290755) [](ancestors = 343290755)